backtrace-library/src/main/java/backtraceio/library/metrics/UniqueEvent.java

```diff
@@ -0,0 +1,52 @@
+package backtraceio.library.metrics;
+
+import com.google.gson.annotations.SerializedName;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import backtraceio.library.logger.BacktraceLogger;
+
+public class UniqueEvent extends Event {
+
+    private final static transient String LOG_TAG = UniqueEvent.class.getSimpleName();
+
+    @SerializedName("unique")
+    private List<String> name;
+
+    UniqueEvent(String name) {
+        this(name, System.currentTimeMillis() / 1000, new HashMap<String, Object>());
+    }
+
+    UniqueEvent(String name, long timestamp, Map<String, Object> attributes) {
+        super(timestamp);
+        this.attributes = attributes;
+        this.name = new ArrayList<String>() {{
+            add(name);
+        }};
+    }
+
+    // The spec specifies name as a JSON array but we don't change the interface yet
+    @Override
+    public String getName() {
+        if (this.name != null && this.name.size() > 0) {
+            return this.name.get(0);
+        }
+        BacktraceLogger.e(LOG_TAG, "Unique Event name must not be null or empty");
+        return new String();
```

Comment:
Never do new `String()`, always `""`. Useless object creation. Or I am a dinosaur and it doesn't matter in modern Java? lol

Reply:
Won't it implicitly create a new `String` anyway since this method has to return a `String`? I don't think it will matter that much since this is the exception path anyway.

Reply:
"" => new String() - this is a syntax sugar. We should follow the pattern that other java developers use/Bartosz started using in the library.

Reply:
Ah I see Bartosz is using this pattern and it is standard Java practice, thank you @vlussenburg @konraddysput !